armory/utils/export.py

```diff
@@ -1,74 +1,51 @@
 import os
 import logging
+import abc
 import numpy as np
 import ffmpeg
 import pickle
 import time
 from PIL import Image, ImageDraw
 from scipy.io import wavfile
 
-from armory.data.datasets import ImageContext, VideoContext, AudioContext, So2SatContext
-
 
 logger = logging.getLogger(__name__)
 
 
 class SampleExporter:
-    def __init__(self, base_output_dir, context, num_samples):
+    def __init__(self, base_output_dir, num_samples):
         self.base_output_dir = base_output_dir
-        self.context = context
         self.num_samples = num_samples
```

Comment:
Does it make more sense for `num_samples` to be part of the scenario base class instead of the exporter? That would simplify its functionality and make it easier to test.

Reply:
I do think this would make the export code simpler. Somewhere we have to keep track and ensure we don't export more than the specified number of samples in the config value. We loop through the different elements of the batch inside the `Exporter` code, rather than the scenario code, so it's a little easier to keep track of this in the `Exporter` code with an `Exporter` instance attribute. 

But it can be done from the scenario code too. I added commit dcc11ac to address this. What I don't love about this change is having `export_samples()` take a positional arg of `num_samples`. Thoughts? Could also have this kwarg default to `None`, in which case all samples in the batch are exported. This way a user could simply call `self.export_samples()` in an interactive session

What I'm trying to avoid is, for ex., you set `batch_size` to 64 but `export_samples` to 1, you want 1 sample exported not 64. This would be a little simpler to handle if it was `export_batches` instead

Reply:
I like the idea of using `export_batches`. Since we tend to think in terms of batches (e.g., `--num-eval-batches`) and since we usually set `batch_size=1` anyway, I think that would be a nice simplification.

Reply:
made this switch in 7cfbd82